---
title: "data-analysis-replication"
author: "Jessica V"
date: "`r Sys.Date()`"
output: html_document
---

# *Introduction*

*OBJECTIVE:*

The overall objective of Legendre et al., 2016 was to estimate metabolic rates of fossil archosauromorphs (a major group of diapsids, differentiated from the other diapsids by the presence of single openings in each side of the skull, in front of the eyes, among other characteristics) by performing a statistical predictive model using a new method of phylogenetic eigenvector maps on a set of bone histological features for a sample of extant and extinct vertebrates.


TYPES OF *DATA COLLECTED:*

The models were built using 57 specimens belonging to 14 extant species and 14 extinct species of tetrapods.

Resting metabolic rates (RMR values) are taken from Montes et al. (2007), where growth rate was taken from juvinile specimens in vivo. (All variables, such as histological characters and metabolic were taken from the same specimen.) Bone histology sections were used for obtaining the following measurements: vascular density, osteocyte density, osteocyte shape, and osteocyte area—on each bone.

Phylogeny relationships, topology and branch length, a well as inner relationships were all taken from published sources (Cubo et al. (2012), Legendre et al. (2013), Benton et al. 2015), Conrad (2008), Joyce et al. (2013),  Nesbitt (2011), and Crawford et al. (2015)).


The *MAIN RESULTS* of the paper showed:

1) that Mesozoic theropod dinosaurs exhibit metabolic rates very close to those found in modern birds
2) that archosaurs share a higher ancestral metabolic rate than that of extant ectotherms
3)that this derived high metabolic rate was acquired at a much more inclusive level of the phylogenetic tree, among non-archosaurian archosauromorphs.

Additionally, these results highlight the difficulties of assigning a given heat production strategy, such as endothermy or ectothermy, to an estimated metabolic rate value. The results of the paper also confirmed findings from previous studies which state that the definition of the endotherm/ectotherm dichotomy may be ambiguous.

*ANALYSIS* DONE:


The PEM approach was selected to estimate mass-specific RMR (in mL O2 h−1 g−0.67 ) in this study, using bone histological data and phylogenetic information to convert the phylogeny into PEMs and predict values of the dependent variable for fossil species.
Three predictive models were done for each of the three elements (femur, humerus, and tibia), to maintain a strict frame of homology.
Ancestral states of RMR were reconstructed from both observed and predicted values, using a phylogeny of the complete sample.

*ANALYSIS REPLICATED*

Visualization:

Here I replicated the sepratate components of Figure 1. Separate images have been coded below as the original work was tied together using a program other than R. 

statistical analysis,  one inferential statistical analysis:

The data analysis I replicated the was done in order to reconstruct the ancestral states of RMR from both observed and predicted values by using a phylogeny of the complete sample. 

#Ancestral RMR values were estimated for all nodes using the function fastAnc from the R package phytools


These are the packages that I used for this replication:
```{r}
library(TreeTools)
library(ape)
library(ggplot2)
library(tidyverse)
library(ggtree)
library(BiocManager)
library(ggtree)
library(phytools)
library(maps)
library(gridExtra)
library(MPSEM)
library(evobiR)
library(Metrics)
```

## *Visualization of Data*

Legendre et al., 2016 has made their data available at "https://datadryad.org/stash/dataset/doi:10.5061/dryad.2853k".

The following data files were downloaded from Dryad to my computer: femurfilemeta.txt, humerusfilemeta.txt, tibiafilemeta.txt, Treecomplete.trees.nex, TreefemurMeta.nex,TreehumerusMeta.nex, TreetibiaMeta.nex.


The files were loaded into R using the the "Files" tab and finding the files through my directory as indicated below. I titled each data by the bone it refers to and viewed the head of the data.



```{r}

f<- "/Users/jessicavaldes/Desktop/R2024/FINAL.PROJECT/tibiafilemeta.txt"

tibia <- read.delim(f, stringsAsFactors = FALSE, fileEncoding = "latin1")

head(tibia)

f<- "/Users/jessicavaldes/Desktop/R2024/FINAL.PROJECT/humerusfilemeta.txt"

humerus <- read.delim(f, stringsAsFactors = FALSE, fileEncoding = "latin1")

head(humerus)

f<- "/Users/jessicavaldes/Desktop/R2024/FINAL.PROJECT/femurfilemeta.txt"

femur <- read.delim(f, stringsAsFactors = FALSE, fileEncoding = "latin1")

head(femur)
```

Here I have loaded the tree data  by reading the NEXUS file and naming them by the component (tibia, humerus, femus, or complete) and adding "tree" in the name.

Additional trees I have replicated that are not found in any figures on the paper can be found below in the "Extra" section.

```{r}
tibiatree <- read.nexus(file = "/Users/jessicavaldes/Desktop/R2024/FINAL.PROJECT/TreetibiaMeta.nex")

head(tibiatree)

humerustree <- read.nexus(file = "/Users/jessicavaldes/Desktop/R2024/FINAL.PROJECT/TreehumerusMeta.nex")

head(humerustree)

femurtree <- read.nexus(file = "/Users/jessicavaldes/Desktop/R2024/FINAL.PROJECT/TreefemurMeta.nex")

head(femurtree)

completetree <- read.nexus(file =   "/Users/jessicavaldes/Desktop/R2024/FINAL.PROJECT/Treecomplete.trees.nex")

head(completetree)


```

COMPLETE TREE:
Here I start replicating the complete tree show in in Figure 1 of the paper.
I made sure the NEXUS file was loaded.

I converter the complete tree into a phylo object.

I plotted the phylo tree using ggtree, add tip labels (shows the species name in this case) and apply a tree-like theme, and name it Main Tree

```{r}

completetree <- read.nexus(file = "/Users/jessicavaldes/Desktop/R2024/FINAL.PROJECT/Treecomplete.trees.nex")

phylocomplete <- as.phylo(completetree)


Maintree <- ggtree(phylocomplete) +
  geom_tiplab() +  
  theme_tree() 

plot(Maintree)

```

Although the Maintree produced has the correct data, it does not look like the image in Figure 1 yet.

Here I rotate each of the nodes individually. (Note: I use the phylo  object in order to be able to rotate the nodes using the rotate.multi function). I name the correctly rotated tree L1.

I visualize the tree as a Cladogram (or rectangular layout) as the paper does by modifying the lengths of the branches from the tree and name it LL1.

```{r}
plot(phylocomplete)

L1<- rotate.multi(phylocomplete, 29)

L1<- rotate.multi(L1, 30)
L1<- rotate.multi(L1, 31)
L1<- rotate.multi(L1, 32)
L1<- rotate.multi(L1, 33)
L1<- rotate.multi(L1, 34)
L1<- rotate.multi(L1, 35)
L1<- rotate.multi(L1, 36)
L1<- rotate.multi(L1, 37)
L1<- rotate.multi(L1, 38)
L1<- rotate.multi(L1, 39)
L1<- rotate.multi(L1, 40)
L1<- rotate.multi(L1, 41)
L1<- rotate.multi(L1, 42)
L1<- rotate.multi(L1, 43)
L1<- rotate.multi(L1, 44)
L1<- rotate.multi(L1, 45)
L1<- rotate.multi(L1, 46)
L1<- rotate.multi(L1, 47)
L1<- rotate.multi(L1, 48)
L1<- rotate.multi(L1, 49)
L1<- rotate.multi(L1, 50)
L1<- rotate.multi(L1, 51)
L1<- rotate.multi(L1, 52)
L1<- rotate.multi(L1, 53)
L1<- rotate.multi(L1, 54)
L1<- rotate.multi(L1, 55)


plot(L1)

LL1<-compute.brlen(L1, length =1)

plot(LL1)
```

Now that the tree is replicated I am moving onto the right part of Figure 1 (for the purposes of this report, I am calling this image replication Part 2).

For Part 2, I need to combine all of the RMR of all of the species provided with the three sets of data (tibia, humerus, and femur.)

In order to make my life easier when combining this data I start by changing the names of the Resting Metabolic Rate columns to "RMR". I then check the column names to make sure I have the correct new column name titles.

```{r}
colnames(tibia)[2] <- "RMR"
colnames(humerus)[2] <- "RMR"
colnames(femur)[2] <- "RMR"

colnames(tibia)
colnames(humerus)
colnames(femur)
```

Here I combine all RMR from all bone datasets.
I combine all Species from all bone datasets. 
I cbind Species and RMR, removing NA's and duplicates, and name the final dataset allSpeciesRMR1.

```{r}
allRMR <- data.frame(RMR = c(femur$RMR, tibia$RMR, humerus$RMR))
allSpecies <- data.frame(Species = c(femur$Species, tibia$Species, humerus$Species))
allsSpeciesRMR <- cbind(allSpecies, allRMR)

allSpeciesRMR <- drop_na(allsSpeciesRMR)

allSpeciesRMR1 <- allSpeciesRMR %>% distinct(Species, .keep_all = TRUE)
```
Now that I have all of the data together I can generate the plot.
Here I plot RMR on the x-axis and Species on the Y-axis
I make sure the shapes are squares (shape 15)
I makes values less than 1 blue, and values more than 1 red.


```{r}


firstplot <- (ggplot(allSpeciesRMR1, aes(x = RMR, y = Species)) + 
  geom_point(shape = 15, size = 4, aes(fill = RMR, colour = ifelse(RMR < 1, "blue", "red"))) +
  scale_colour_identity() +
  scale_fill_identity() +
    scale_fill_gradient(low = "red", high = "red", breaks = c(0, 1, 3), labels = c("0", "1", "3"))+
  guides(colour = FALSE) + 
 theme_bw() + 
  theme(legend.position="bottom", legend.key.size = unit(.5, "cm"),
  legend.key.width = unit(2,"cm")))

plot(firstplot)

```


I then realize that the data above only contains the data provided in the data from dryad.

In order to have all of the data reported on Figure 1, I have manually added all of the Table 1 data from the paper below(Species, RMR, UL, and LL), and cbind them into a dataframe called missingspecies. I then convert to the same type of data and join this data with the previous data above into a dataframe called full data.
```{r}
Species <- c("Prolacerta broomi", "Proterosuchus fergusi", "Garjainia madiba", "Euparkeria capensis", "Rutiodon carolinensis", "Calyptosuchus wellesi", "Postosuchus kirkpatricki", "Lesothosaurus diagnosticus", "Maiasaura peeblesorum", "Thecodontosaurus antiquus", "Coelophysis bauri", "Lourinhanosaurus antunesi", "Allosaurus fragilis", "Troodon formosus")
RMR<- c(1.779574478, 0.742796421, 1.428587942, 1.388275876, 0.395353151, 1.707702381, 1.864455587, 1.130079749, 0.861329988, 1.002164215, 11.83642213, 2.285600254, 5.889620032, 3.301842183)
LL <- c(1.319254894,0.604057894, 1.093892708, 1.05448569, 0.308843779, 1.135032404, 1.165933134, 0.78543557, 0.532978612, 0.760912118, 7.44819231, 1.624566141, 3.680724942, 2.388193685)
UL <- c(2.401309306, 0.914306682, 1.865918168, 1.827725139, 0.506094433, 2.569307738, 2.981470097, 1.630369802, 1.400681483, 1.319906895, 18.81005185, 3.215608472, 9.478280852, 4.56502413)

missingspecies <- cbind(Species, RMR, LL, UL)
missingspecies <- as.data.frame(missingspecies)
colnames(missingspecies)
colnames(allSpeciesRMR1)
class(allSpeciesRMR1)
class(missingspecies)


missingspecies <- mutate(missingspecies, RMR = as.double(RMR))  

fulldata<- (full_join(allSpeciesRMR1, missingspecies, by = c("RMR", "Species")))

fulldata
```

The data is now ready to be fully plotted.

Here I plot the full dataset,  using shapes and colors like the paper.
I understand that this paper was published before R packages were as advanced as they are now, thus the author created images outside of R. So this image has more similar colors to that of the image created outside of R.

In order to match the original image better, I change "blue" to "light blue"
#Add a title to the x-axis
#add a scale gradient to the legend along with the breaks the paper has.
#remove the grey background
```{r}

plot2 <-(ggplot(fulldata, aes(x = RMR, y = Species)) +
  geom_point(shape = 15, size = 4, aes(fill = RMR, colour = ifelse(RMR < 1, "light blue", "red"))) +
  labs( x = "Mass-specific resting metabolic rate") +
scale_colour_identity() +
  scale_fill_identity() +
    scale_fill_gradient(low = "light blue", high = "red", breaks = c(0, 1, 2, 3, 6, 8, 10, 12), labels = c("0", "1", "2", "3", "6", "8", "10", "12"))+
  guides(colour = FALSE) + 
 theme_bw() + 
  theme(legend.position="bottom", legend.key.size = unit(.5, "cm"),
  legend.key.width = unit(2,"cm")))

plot(plot2)
```

Here I have created a second image using fill gradients 
I changed the shape to 22 as the squares are outlined in the paper and save image as FIGURE1


```{r}

FIGURE1 <- (ggplot(fulldata, aes(x = RMR, y = Species)) + 
   geom_point(shape = 22, size = 4, color = "black", aes(fill = RMR, colour = ifelse(RMR < 1, "light blue", "red"))) +
   labs( x = "Mass-specific resting metabolic rate") +
  scale_colour_identity() +
    scale_fill_identity() +
     scale_fill_gradient(low = "light blue", high = "red", breaks = c(0, 1, 2, 3, 6, 8, 10, 12), labels = c("0", "1", "2", "3", "6", "8", "10", "12"))+
   guides(colour = FALSE) + 
  theme_classic() + 
  theme(plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), 
                                "inches")) +
    theme(legend.position="bottom", legend.key.size = unit(.1, "inches"),
   legend.key.width = unit(1,"inches")))

plot(FIGURE1)
```

I made sure RMR, UL, amd LL were numeric and added the errorbars to the plot naming it fixingscale.
```{r}
fulldata$RMR<-as.numeric(fulldata$RMR)
fulldata$UL<-as.numeric(fulldata$UL)
fulldata$LL<-as.numeric(fulldata$LL)


fixingscale <- (ggplot(fulldata , aes(x = RMR, y = Species)) + 
   geom_point(shape = 22, size = 4, color = "black", aes(fill = RMR)) +
   labs( x = "Mass-specific resting metabolic rate") +
     scale_fill_gradient(low = "light blue", high = "red", breaks = c(0, 1, 2, 3, 6, 8, 10, 12), labels = c("0", "1", "2", "3", "6", "8", "10", "12"))+
  theme_classic() + 
    geom_errorbarh(aes(xmax=UL, xmin=LL)) +
  theme(plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "inches")) +
    theme(legend.position="bottom", legend.key.size = unit(.1, "inches"),
   legend.key.width = unit(1,"inches")))



plot(fixingscale)
  

```


```{r}
#Here I filder the full data to remove NA's



ggplot(fulldata, aes(x = RMR, y = Species)) + 
   geom_point(shape = 22, size = 4, color = "black", aes(fill = RMR, colour = ifelse(RMR < 1, "light blue", "red"))) +
   labs( x = "Mass-specific resting metabolic rate") +
  scale_colour_identity() +
    scale_fill_identity() +
     scale_fill_gradient(low = "light blue", high = "red", breaks = c(0, 1, 2, 3, 6, 8, 10, 12), labels = c("0", "1", "2", "3", "6", "8", "10", "12"))+
   guides(colour = FALSE) + 
  theme_classic() + 
  theme(plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), 
                                "inches")) +
    theme(legend.position="bottom", legend.key.size = unit(.1, "inches"),
   legend.key.width = unit(1,"inches"))

#making sure to take out any NA's and I created a new dataset called error bars for that

#Species order

#Here I create the species order found on Figure 1 Part 2.
```

Here I made sure to use the order in which the species appear on the paper. I manually added the order as Sorder2 and ploted the final plot.

```{r}

Sorder2 <- c("Gallus gallus", "Anas platyrhynchos",  "Troodon formosus", "Lourinhanosaurus antunesi","Allosaurus fragilis","Coelophysis bauri","Thecodontosaurus antiquus", "Maiasaura peeblesorum", "Lesothosaurus diagnosticus", "Calyptosuchus wellesi","Postosuchus kirkpatricki", "Crocodylus niloticus", "Rutiodon carolinensis",  "Euparkeria capensis","Garjainia madiba","Proterosuchus fergusi", "Prolacerta broomi", "Trachemys scripta", "Pelodiscus sinensis","Chelodina oblonga","Varanus niloticus","Varanus exanthematicus","Podarcis muralis","Zootoca vivipara", "Mus musculus", "Cavia porcellus", "Microcebus murinus","Pleurodeles waltl")

fulldata$Species <- factor(fulldata$Species, levels=Sorder2)

final <- (ggplot(fulldata , aes(x = RMR, y = Species)) + 
   geom_point(shape = 22, size = 4, color = "black", aes(fill = RMR)) +
   labs( x = "Mass-specific resting metabolic rate") +
     scale_fill_gradient(low = "light blue", high = "red", breaks = c(0, 1, 2, 3, 6, 8, 10, 12), labels = c("0", "1", "2", "3", "6", "8", "10", "12"))+
  theme_classic() + 
    geom_errorbarh(aes(xmax=UL, xmin=LL)) +
   theme(plot.margin = unit(c(0.5, 0.5, 0.5, 0.5), "inches")) +
    theme(legend.position="bottom", legend.key.size = unit(.1, "inches"),
   legend.key.width = unit(1,"inches")))


plot(final)

```

ANALYSIS REPLICATION

## Statistical Replications/Reanalysis

[Be sure to **thoroughly** explain what replications you are doing and comment your code so that it is easy for a reader to understand. Include in this section relevant tables/figures/values from the original paper for comparison to what you accomplished with your replication. Note that I want you to do the bulk of any *exposition* using text and markdown syntax outside of code blocks. That is, your document should not just be one big code block with ***R*** style comments but rather a nicely formatted report with code separated from exposition, interpretation, and dicussion.]

###To investigate further into the evolution of high resting metabolic rates in amniotes, we reconstructed ancestral states of RMR from both observed and predicted values, using a phylogeny of our complete sample. 

Ancestral RMR values were estimated for all nodes using the function fastAnc from the R package phytools (Revell 2012); this function compiles maximum likelihood ancestral states with 95% confidence intervals assuming a Brownian motion model, and allows us to plot it on the original phylogeny (Fig. 2).



## Summary/Discussion

[Narrative section that overviews how successful were you at replicating the analyses and visualizations in the study. What problems did you encounter? Why might you have encountered those problems? What details were lacking from the original study's methods that might have hampered your ability to replicate the authors' results?]

## References

[Include the citation for your paper, plus any other literature you might reference.]

Create a new RStudio project detailing your work on replicating the results presented in the paper you choose. 

Note that I will be looking for you to clearly take your reader through all of the elements of data manipulation, analysis, and, if appropriate, visualization. You should provide as much coding detail, explanation, and output tables as necessary to compare your results to those published!

You do not need to replicate ALL of the analyses presented in the paper (although the more the better)!

As assessment, I will be looking at several different elements of your report and code for this assignment. Below, I outline some of the main things:

Organization and Logistics

Repo set up on GitHub, named correctly, and shared with me (URL submitted via Canvas)
PDF of paper included in the top level of the repository
“.Rmd” or “.qmd” file for your report stored at the top level of the repo
“.Rmd” or “.qmd” file includes text and code and embedded images/tables from the paper for comparison
“.Rmd” or “.qmd” file well organized into subsections, including blocks of explanatory text and R code blocks, plus output
R code follows a consistent convention with respect to variable and function names and formatting
File(s) with original data included in a directory (folder) called “data” within the repo
Images from the original paper that are referenced in the report stored in a separate directory called “images” within the repo
Introduction and Framing

Report includes a short, introductory description of the goal of the original paper, the data used, the analyses conducted, and the conclusions of the original study
Report outlines the specific data and reanalyses you will be doing
Code correctly loads all required packages not included in {base} R
“.Rmd” or “.qmd” file renders and produces HTML output without requiring edits or additional modifications
“.Rmd” or “.qmd” file defines variable names used in R code
R code successfully reads in data file(s) from the local “data” directory within the repo and shows a few lines of raw data (e.g., using head())
R code successfully reads in any image file(s) for comparison from within the local “image” directory within the repo
Data Analysis/Visualization Replications

For each of the analyses/visualizations being done…

Text of the report clearly takes the reader through all of the elements of data manipulation/analysis/visualization, from raw data to presentation
Report text is thorough and the R code is well-documented and provides as much explanation and output (tables, figures, etc.) as necessary to understand how the code works
Report includes side-by-side comparisons of the results of the replication attempts to the published original results
Discussion and Reflection

Report includes a narrative summary about how successful the analysis replications were or were not - i.e., how well do the replicated results compare to those presented in the original paper?
Report discusses and addresses any challenges encountered… missing data, unclear information about how data were processed in the original publication, etc.
Report discusses where and possible reasons why the analysis replications might differ from the authors’ original results



we reconstructed ancestral states of RMR from both observed and predicted values, 

using a phylogeny of our complete sample. 

Ancestral RMR values were estimated for all nodes using the function 

fastAnc from the R package phytools (Revell 2012); this function compiles maximum likelihood ancestral states with 95% confidence intervals assuming a Brownian motion model, and

## Load anole tree
anole.tree <- read.tree("http://www.phytools.org/eqg2015/data/anole.tre")

## Load anole trait data, extract snout-vent-length (svl) as named vector
svl <- fulldata %>%
  mutate(svl = set_names(svl, Species)) %>%
  pull(svl)

ssvl <- tibia %>%
  mutate(svl=set_names(svl, Species)) %>%
  pull(svl)


svl
head(svl)
# Plot with default color scheme
contmap_obj <- contMap(phylocomplete, svl, plot = FALSE)

plot(
  contmap_obj, 
  type="fan", 
  legend = 0.7*max(nodeHeights(anole.tree)),
  fsize = c(0.5, 0.7))
nole.tree <- read.tree("http://www.phytools.org/eqg2015/data/anole.tre")

head(anole.tree)

obj<-contMap(phylocomplete, svl,plot=FALSE)

head(completetree)

plot(obj,lwd=7,xlim=c(-0.2,3.6))
errorbar.contMap(obj)




svl <- read.csv(t, row.names=1)
9
svl <- as.matrix(phylocomplete$tip.label)[,1] 

fit <- fastAnc(phylocomplete,svl,vars=TRUE,CI=TRUE)



fit_obj <- contMap(anole_tree, svl, plot=FALSE)
plot(fit_obj, type="fan", legend=0.7*max(nodeHeights(anole_tree)), fsize=c(0.7,0.9))
#convert dataframe to a vector

## load data from Garland et al. (1992)
data(mammal.tree)
data(mammal.data)
## extract character of interest
tipnames<-(setNames(phylocomplete$tip.label,
    rownames(phylocomplete)))

tipnames

## estimate ancestral body sizes
Ancestral<-fastAnc(phylocomplete,tipnames,vars = FALSE, CI=TRUE)
print(fit.BM,printlen=10)

phylocomplete[ is.na(phylocomplete) ] <- 0
phylocomplete$tip.label<- apply(phylocomplete,1,function(x) as.numeric(paste0(x , collapse="")))

phylocomplete[ is.na(phylocomplete) ] <- 0
phylocomplete$tip.label <- as.numeric(do.call(paste0, phylocomplete))

head(completetree)

if(any(is.na(phylocomplete$tip.label))){
  stop("NA values found in phylocomplete$tip.label. Please ensure all tip labels are non- NA.")}

drop.tip(phylocomplete, phylocomplete$tip.label, trim.internal = TRUE)

tib_label<- get_tip_labels(phylocomplete)

tipnames<-TipLabels(phylocomplete, single=TRUE)

tip_labelss <- phylocomplete$tip.label[!is.na(phylocomplete$tip.label)]

ancestral_states <- fastAnc(phylocomplete, tip.label, vars = FALSE, CI = FALSE)

head(phylocomplete)


     ancestral_states <- fastAnc(completetree, completetree$tip.label,names(RMR), vars = FALSE, CI = FALSE)
ancestral_states <- fastAnc(phylocomplete, phylocomplete$tip.label, vars = FALSE, CI = FALSE)

class(phylocomplete$tip.label)

setdiff(phylocomplete$tip.label)
setdiff(completetree$tip.label,names(RMR))
#fastAnc(phylocomplete,phylocomplete$edge.length, vars=FALSE, CI=FALSE)

#edge_lengths <- setNames(phylocomplete$edge.length, phylocomplete$tip.label)

ancestral_states <- fastAnc(phylocomplete, RMR, vars = FALSE, CI = FALSE)

#edge_lengths <- edge_lengths[match(phylocomplete$tip.label, names(edge_lengths))]

#ancestral_states <- fastAnc(phylocomplete, edge_lengths, vars = FALSE, CI = FALSE)

#head(ancestral_states)

#colnames(ancestral_states)
## extract character of interest
#fastAn <-log(setNames(phylocomplete$edge.legth,
 #   rownames(phylocomplete)))


#class(phylocomplete$edge.length)
## estimate ancestral body sizes
#fit.BM<-fastAnc(Maintree,ln.bodyMass,CI=TRUE)
#print(fit.BM,printlen=10)

#fastAnc(phylocomplete, x_ordered, vars = FALSE, CI = FALSE)

#str(phylocomplete)
#x_ordered <- phylocomplete$tip.label[match(phylocomplete, phylocomplete$tip.label)]


#ancestralstates <- fastAnc(complete, complete$tip.label)

#Extra
This section contains extra replications I have done for this project.
I decided to replicate these trees although they were not in any figures on the paper because I thought it was a 

```{r}
t <- "/Users/jessicavaldes/Desktop/R2024/FINAL.PROJECT/TreefemurMeta.nex"


# Here I read the NEXUS file
femurtree <- read.nexus(file = "/Users/jessicavaldes/Desktop/R2024/FINAL.PROJECT/TreefemurMeta.nex")
# Here I convert the femurtree to a phylo object
phylofemurtree <- as.phylo(femurtree)

#BiocManager::install("ggtree")

# Plot the phylo tree using ggtree, add tip labels and pply a tree-like theme
ggtree(phylofemurtree) +
  geom_tiplab() +  
  theme_tree()  


```


```{r}

t <- "/Users/jessicavaldes/Desktop/R2024/FINAL.PROJECT/TreefemurMeta.nex"


# Here I read the NEXUS file
femurtree <- read.nexus(file = "/Users/jessicavaldes/Desktop/R2024/FINAL.PROJECT/TreefemurMeta.nex")
# Here I convert the femurtree to a phylo object
phylofemurtree <- as.phylo(femurtree)

#BiocManager::install("ggtree")

# Plot the phylo tree using ggtree, add tip labels and pply a tree-like theme
ggtree(phylofemurtree) +
  geom_tiplab() +  
  theme_tree()     

plot(femurtree)
plot(femurtree, use.edge.length=FALSE)
nodelabels()

femurtree<- rotate.multi(femurtree, 22)
femurtree<- rotate.multi(femurtree, 23)
femurtree<- rotate.multi(femurtree, 24)
femurtree<- rotate.multi(femurtree, 25)
femurtree<- rotate.multi(femurtree, 26)
femurtree<- rotate.multi(femurtree, 27)
femurtree<- rotate.multi(femurtree, 28)
femurtree<- rotate.multi(femurtree, 29)
femurtree<- rotate.multi(femurtree, 30)
femurtree<- rotate.multi(femurtree, 31)
femurtree<- rotate.multi(femurtree, 32)
femurtree<- rotate.multi(femurtree, 33)
femurtree<- rotate.multi(femurtree, 34)
femurtree<- rotate.multi(femurtree, 35)
femurtree<- rotate.multi(femurtree, 36)
femurtree<- rotate.multi(femurtree, 37)
femurtree<- rotate.multi(femurtree, 38)
femurtree<- rotate.multi(femurtree, 39)
femurtree<- rotate.multi(femurtree, 40)
femurtree<- rotate.multi(femurtree, 41)
```

TREE 2

Humerus tree shown here.
```{r}
humerustree <- read.nexus(file = "/Users/jessicavaldes/Desktop/R2024/FINAL.PROJECT/TreehumerusMeta.nex")

phylohumerustree <- as.phylo(humerustree)


ggtree(phylohumerustree) +
  geom_tiplab() +  
  theme_tree()   

```
TREE TIBIA

```{r}
tibiatree <- read.nexus(file = "/Users/jessicavaldes/Desktop/R2024/FINAL.PROJECT/TreetibiaMeta.nex")

phylotibiatree <- as.phylo(tibiatree)


ggtree(phylotibiatree) +
  geom_tiplab() +  
  theme_tree()   
```


PEM

#```{r}

#in order to extract PEM first I

# Extract eigenvectors from the tree
grloc <- getGraphLocations(LucasTree,phylocomplete[is.na(data[,"RMR"]),"species"])

head(grloc)

sporder <- match(attr(grloc$x,"vlabel")[grloc$x$vertex$species],phylocomplete[,"species"])

#then I build the PME, here is a list of variables

aux <- c("Vasc_density","Cell_density","Cell_size","Cell_shape")

1+1

PEMfs <- list()
  for(m in aux) {
    PEMfs[[m]] <- PEM.fitSimple(y=data[sporder,"RMR"],
                                        x=data[sporder,m],w=grloc$x,d="distance",sp="species",lower=0,upper=1)
  PEMfs[["none"]] <- PEM.fitSimple(y=data[sporder,"RMR"],
                                           x=NULL,w=grloc$x,d="distance",sp="species",lower=0,upper=1)
} ; rm(m)

```
RESTART
#```{r}
f<- "/Users/jessicavaldes/Desktop/R2024/FINAL.PROJECT/femurfilemeta.txt"

femur <- read.delim(f, stringsAsFactors = FALSE, fileEncoding = "latin1")

head(femur)c

#Here I read the femus data as a table
#f<- "/Users/jessicavaldes/Desktop/R2024/FINAL.PROJECT/femurfilemeta.txt"

#femurdata<-read.table(f, header=T)

#head(femur)

t <- "/Users/jessicavaldes/Desktop/R2024/FINAL.PROJECT/TreefemurMeta.nex"

#tabledata <- read.table(t, header = T)
# Here I read the NEXUS file
Ftree <- read.nexus(t)
plot(Ftree)

phyloftree <- as.phylo(Ftree)

class(phyloftree)

plot(phyloftree, use.edge.length=FALSE)
nodelabels()

finaltree <- ReorderData(Ftree, femur, taxa.names = "Species")

plot(finaltree)
 
femurtree2 <- read.tree(f)

newfemurtree<- ggtree(Ftree, branch.length='none', layout="rectangular") + geom_tiplab() 

plot(newfemurtree)

rotateNodes(newfemurtree, 29)



# Match the order of species in the dataset with that in the tree
reorderedtree <- ReorderData(Ftree, femur, taxa.names = "Species")






plot(reorderedtree)

phyloreordered <- as.phylo(reorderedtree)

plotTree(reordered)


newfemur <- femur[1]<-femur[1]; femur[,c(2,4:6)]<-log(femur[,c(2,4:6)]); femur[,3]<-log1p(femur[,3])

 ###
# Here I convert the femurtree to a phylo object
phylofemurtree <- as.phylo(femurtree)

#BiocManager::install("ggtree")

# Plot the phylo tree using ggtree, add tip labels and pply a tree-like theme
ggtree(phylofemurtree) +
  geom_tiplab() +  
  theme_tree()   
```


#```{r}

### PEM ###

# Extract eigenvectors from the tree

#Here I make the collumnb title easier into RMR
colnames(femur)[2] <- "RMR"
colnames(femur)[3] <- "vascular.density"
colnames(femur)[4] <- "ost.density"
colnames(femur)[5] <- "ost.area"
colnames(femur)[6] <- "ost.shape"
#here I log transform the data in columns 2,4,5,6  and # log1p(x) = ln(1+x) – convenient for null or very small values for collumn 3

femur1 <- femur[,c(2,4:6)]<-log(femur[,c(2,4:6)]); femur[,3]<-log1p(femur[,3])

Species <- (femur$Species)
RMR <- log(femur$RMR)
logVD <- log1p(femur$vascular.density)
logROD <- log(femur$ost.density)
logOA <- log(femur$ost.area)
logOS <- log(femur$ost.shape)

#here I combine all transformed data
FEMURDATA <- cbind(Species, RMR, logVD, logROD, logOA, logOS)





colnames(editedfemur)[1] <- "RMR"

grloc <- getGraphLocations(Ftree,editedfemur[is.na(editedfemur[,"RMR"]),"Species"])



sporder <- match(attr(grloc$x,"vlabel")[grloc$x$vertex$Species],editedfemur[,"Species"])




```


#```{r}
PEM

sum(is.na(FEMURDATA[, 1]))
# Extract eigenvectors from the tree
grloc <- getGraphLocations(LucasTree,FEMURDATA[is.na(FEMURDATA[,"RMR"]),"Species"])

print(dim(FEMURDATA))
subset_data <- FEMURDATA[is.na(FEMURDATA[,"RMR"]),"Species"]
print(dim(subset_data))
grloc <- getGraphLocations(LucasTree, subset_data)



sporder <- match(attr(grloc$x,"vlabel")[grloc$x$vertex$species],data[,"species"])

class(FEMURDATA)
```


Lucas
#```{r}

data<-read.delim(f, stringsAsFactors = FALSE, fileEncoding = "latin1")
colnames(data)[2] <- "RMR"
head(data)

tree<- read.nexus(t)

plot(tree)
head(complete)
grloc <- getGraphLocations(phylofemurtree,data[is.na(data[,"RMR"]),"Species"])

phylofemurtree<- as.phylo(femurtree)

head(femurtree)
str(complete)

SpecieswithNA <- c("Calyptosuchus wellesi", "Lesothosaurus diagnosticus",	
"Troodon formosus", "Allosaurus fragilis","Rutiodon carolinensis", "Maiasaura peeblesorum",	
"Proterosuchus fergusi")
grloc <- getGraphLocations(complete, FEMURDATA)


grloc <- getGraphLocations(LucasTree,FEMURDATA[,"RMR","Species"])



str(tpall$Nnode)

head(grloc)

sporder <- match(attr(grloc$x,"vlabel")[grloc$x$vertex$Species],FEMURDATA[,"Species"])

```
[Include a view of the first few lines of the dataset plus any exploratory data analysis - e.g., overview of descriptive statistics included in the paper, plots, etc.]

# *Discussion and Reflection*

Report includes a narrative summary about how successful the analysis replications were or were not - i.e., how well do the replicated results compare to those presented in the original paper?

Report discusses and addresses any challenges encountered… missing data, unclear information about how data were processed in the original publication, etc.

Report discusses where and possible reasons why the analysis replications might differ from the authors’ original results